src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp

```diff
@@ -597,6 +659,666 @@ HRESULT STDMETHODCALLTYPE CorProfiler::JITCompilationStarted(
   return S_OK;
 }
 
+//
+// ICorProfilerCallback6 methods
+//
+HRESULT STDMETHODCALLTYPE CorProfiler::GetAssemblyReferences(
+    const WCHAR* wszAssemblyPath,
+    ICorProfilerAssemblyReferenceProvider* pAsmRefProvider) {
+
+  auto assemblyPathString = ToString(wszAssemblyPath);
+  auto filename =
+      assemblyPathString.substr(assemblyPathString.find_last_of("\\/") + 1);
+  auto lastNiDllPeriodIndex = filename.rfind(".ni.dll");
+  auto lastDllPeriodIndex = filename.rfind(".dll");
+  if (lastNiDllPeriodIndex != std::string::npos) {
+    filename.erase(lastNiDllPeriodIndex, 7);
+  } else if (lastDllPeriodIndex != std::string::npos) {
+    filename.erase(lastDllPeriodIndex, 4);
+  }
+
+  // Get assembly name
+  const WSTRING assembly_name = ToWSTRING(filename);
+
+  // We must never try to add assembly references to
+  // mscorlib or netstandard. Skip other known assemblies.
+  for (auto&& skip_assembly : skip_assemblies) {
+    if (assembly_name == skip_assembly) {
+      Debug("GetAssemblyReferences skipping known assembly: ", wszAssemblyPath);
+      return S_OK;
+    }
+  }
+
+  std::vector<IntegrationMethod> filtered_integrations =
+      FlattenIntegrations(integrations_);
+
+  // TODO: Make this assembly reference dynamic vs hard-coded
+  const AssemblyReference assemblyReference = trace::AssemblyReference(
+      L"Datadog.Trace.ClrProfiler.Managed, Version=1.6.0.0, Culture="
+      L"neutral, PublicKeyToken=def86d061d0d2eeb");
+
+  ASSEMBLYMETADATA assembly_metadata{};
+  assembly_metadata.usMajorVersion = assemblyReference.version.major;
+  assembly_metadata.usMinorVersion = assemblyReference.version.minor;
+  assembly_metadata.usBuildNumber = assemblyReference.version.build;
+  assembly_metadata.usRevisionNumber = assemblyReference.version.revision;
+  if (assemblyReference.locale == "neutral"_W) {
+    assembly_metadata.szLocale = nullptr;
+    assembly_metadata.cbLocale = 0;
+  } else {
+    assembly_metadata.szLocale =
+        const_cast<WCHAR*>(assemblyReference.locale.c_str());
+    assembly_metadata.cbLocale = (DWORD)(assemblyReference.locale.size());
+  }
+
+  DWORD public_key_size = 8;
+  if (assemblyReference.public_key == trace::PublicKey()) {
+    public_key_size = 0;
+  }
+
+  for (auto& i : filtered_integrations) {
+    if (true) {
+      COR_PRF_ASSEMBLY_REFERENCE_INFO asmRefInfo;
```

Comment:
Let's initialize this to default values to be safe.
```suggestion
      COR_PRF_ASSEMBLY_REFERENCE_INFO asmRefInfo{};
```